### PR TITLE
[microNPU] Added checks for out of range shifts

### PR DESCRIPTION
**Pull request by jacobbohlin**
Bugfix, removed an invalid assert.
Added explicit handling of quantization `scale` and `shift` values that are outside the valid range of the NPU.

### CODE REVIEW

**Comment by jacobbohlin** [comment]
@manupa-arm @NicolaLancellotti @ekalda @lhutton1 @mbaret

**Comment by manupa-arm** [comment]
Thanks @jacobbohlin .

Similiar comment to other PR.
Let us a test case, however in this case I think a small end to end test might be meaningful because it is a low-level change, we might want to make sure this is the right change that produces a CS that runs. WDYT ?

**Comment by jacobbohlin** [comment]
@manupa-arm Sounds good, I will add a small end to end test that takes this code path.

File: tests/python/contrib/test_ethosu/test_codegen.py [hunk 1]
```diff
@@ -242,6 +242,93 @@ def representative_dataset():
...
+    activation = "RELU"
+    accel_type = "ethos-u55-128"
+
+    def create_tflite_graph():
+        class Model(tf.Module):
+            @tf.function
+            def tf_function(self, x):
+                # Use tf.nn API to create the model
+                tf_strides = [1, strides[0], strides[1], 1]
+                weights = np.random.uniform(size=[kernel_shape[-1], kernel_shape[1], 2, 2])
```
Comment (ekalda) [inline tests/python/contrib/test_ethosu/test_codegen.py:262]: It doesn't make an effective difference here, but I think `kernel_shape[-1]` and `kernel_shape[1]` refer to the same element

File: tests/python/contrib/test_ethosu/test_codegen.py [hunk 2]
```diff
@@ -242,6 +242,93 @@ def representative_dataset():
...
+    tflite_graph = create_tflite_graph()
+    tflite_model = tflite.Model.Model.GetRootAsModel(tflite_graph, 0)
+
+    relay_module, params = relay.frontend.from_tflite(
+        tflite_model,
+        shape_dict={"input": ifm_shape},
+        dtype_dict={"input": dtype},
+    )
+    mod = partition_for_ethosu(relay_module, params)
+    print(mod)
```
Comment (ekalda) [inline tests/python/contrib/test_ethosu/test_codegen.py:309]: remove print

**Review by ekalda — COMMENTED** [review]
Looks good, just few nits. I wonder if we can use the functions created after Matthew's refactor to shorten the test?

File: tests/python/contrib/test_ethosu/test_codegen.py (hunk 1 above)
Comment (jacobbohlin) [reply tests/python/contrib/test_ethosu/test_codegen.py]: You are right, I fixed this and refactored to use Matthew's functions

**Review by lhutton1 — APPROVED** [review]
Thanks for the fix @jacobbohlin, LGTM!

**Review by manupa-arm — APPROVED** [review]
LGTM!

**Comment by manupa-arm** [comment]
Thanks @jacobbohlin @ekalda @lhutton1. This is merged now!

**Pull request merged**